apps/pay/app/[username]/transaction/page.tsx

```diff
@@ -77,11 +80,35 @@ function TransactionsPage() {
   const [transactions, setTransactions] = useState<Transaction[]>([])
   const [cursor, setCursor] = useState<string | null>(null)
   const [hasMore, setHasMore] = useState<boolean>(false)
+  const [posCompanionInstalled, setPosCompanionInstalled] = useState<boolean>(false)
+  const session = useSession()
 
   const { data, loading, error, fetchMore } = useGetPaginatedTransactionsQuery({
     variables: { first: 10 },
   })
 
+  useEffect(() => {
+    const checkRelatedApps = async () => {
+      try {
+        if ("getInstalledRelatedApps" in navigator) {
+          /* eslint @typescript-eslint/ban-ts-comment: "off" */
+          // @ts-ignore-next-line no-implicit-any error
+          const apps = await navigator.getInstalledRelatedApps()
+          const isAppInstalled = apps.some(
+            (app: { id: string }) => app.id === "com.blink.pos.companion",
+          )
+          setPosCompanionInstalled(isAppInstalled)
+        } else {
+          console.error("getInstalledRelatedApps is not supported in this browser.")
+        }
+      } catch (error) {
+        console.error("Error checking installed related apps:", error)
```

Comment:
dont we have  log library in galoy pay? using console.log/error is not good

Reply:
No, I don't think we have a log library in Pay.

Reply:
please add it, at least a wrapper and if we have otel integrate it please handle error properly

Reply:
will look into it.